# Heart Failure Analysis

In [1]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns
%matplotlib inline
import matplotlib

import warnings
warnings.filterwarnings('ignore')

pd.set_option('display.max_rows', None)
pd.set_option('display.max_columns', None)  
pd.set_option('display.expand_frame_repr', False)
pd.set_option('max_colwidth', -1)

In [2]:
heart_failure_raw_df = pd.read_csv("C:\\Users\\SHETTS60\\Documents\\Python scripts\\Data\\heart_failure_clinical_records_dataset.csv")
heart_failure_df = heart_failure_raw_df.copy()

In [3]:
heart_failure_df.head()

age  anaemia  creatinine_phosphokinase  diabetes  ejection_fraction  high_blood_pressure  platelets  serum_creatinine  serum_sodium  sex  smoking  time  DEATH_EVENT
0  75.0  0        582                       0         20                 1                    265000.00  1.9               130           1    0        4     1          
1  55.0  0        7861                      0         38                 0                    263358.03  1.1               136           1    0        6     1          
2  65.0  0        146                       0         20                 0                    162000.00  1.3               129           1    1        7     1          
3  50.0  1        111                       0         20                 0                    210000.00  1.9               137           1    0        7     1          
4  65.0  1        160                       1         20                 0                    327000.00  2.7               116           0    0        8     1

# Data Preparation and Cleaning

In [4]:
heart_failure_df.describe()

age     anaemia  creatinine_phosphokinase    diabetes  ejection_fraction  high_blood_pressure      platelets  serum_creatinine  serum_sodium         sex    smoking        time  DEATH_EVENT
count  299.000000  299.000000  299.000000                299.000000  299.000000         299.000000           299.000000     299.00000         299.000000    299.000000  299.00000  299.000000  299.00000  
mean   60.833893   0.431438    581.839465                0.418060    38.083612          0.351171             263358.029264  1.39388           136.625418    0.648829    0.32107    130.260870  0.32107    
std    11.894809   0.496107    970.287881                0.494067    11.834841          0.478136             97804.236869   1.03451           4.412477      0.478136    0.46767    77.614208   0.46767    
min    40.000000   0.000000    23.000000                 0.000000    14.000000          0.000000             25100.000000   0.50000           113.000000    0.000000    0.00000    4.000000    0.00000    
25%    51.000000   0.000000    116.500000                0.000000    30.000000          0.000000             212500.000000  0.90000           134.000000    0.000000    0.00000    73.000000   0.00000    
50%    60.000000   0.000000    250.000000                0.000000    38.000000          0.000000             262000.000000  1.10000           137.000000    1.000000    0.00000    115.000000  0.00000    
75%    70.000000   1.000000    582.000000                1.000000    45.000000          1.000000             303500.000000  1.40000           140.000000    1.000000    1.00000    203.000000  1.00000    
max    95.000000   1.000000    7861.000000               1.000000    80.000000          1.000000             850000.000000  9.40000           148.000000    1.000000    1.00000    285.000000  1.00000

In [5]:
heart_failure_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 299 entries, 0 to 298
Data columns (total 13 columns):
 #   Column                    Non-Null Count  Dtype  
---  ------                    --------------  -----  
 0   age                       299 non-null    float64
 1   anaemia                   299 non-null    int64  
 2   creatinine_phosphokinase  299 non-null    int64  
 3   diabetes                  299 non-null    int64  
 4   ejection_fraction         299 non-null    int64  
 5   high_blood_pressure       299 non-null    int64  
 6   platelets                 299 non-null    float64
 7   serum_creatinine          299 non-null    float64
 8   serum_sodium              299 non-null    int64  
 9   sex                       299 non-null    int64  
 10  smoking                   299 non-null    int64  
 11  time                      299 non-null    int64  
 12  DEATH_EVENT               299 non-null    int64  
dtypes: float64(3), int64(10)
memory usage: 30.5 KB


we can clearly see that there are 299 rows and 12 column

In [6]:
#renaming the columns
heart_failure_df.rename(columns ={'DEATH_EVENT':'patient_dead'},inplace = True)

In [10]:
#convert age column into integer
heart_failure_df['age']=heart_failure_df['age'].astype(int)
#convert few columns into Bool for our convenience
heart_failure_df[['anaemia','diabetes','high_blood_pressure','smoking','patient_dead'] ]= heart_failure_df[['anaemia','diabetes','high_blood_pressure','smoking','patient_dead']].astype(bool)

In [12]:
#changing the value of sex column to male or female
heart_failure_df['sex'] = np.where(heart_failure_df['sex'] == 1, 'Male','Female')

In [15]:
#to convert the platelet-count into kilo-platelets/mcL 
heart_failure_df['platelets'] = (heart_failure_df['platelets'] / 1000)
heart_failure_df['platelets']  = heart_failure_df['platelets'].astype(int)

In [11]:
heart_failure_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 299 entries, 0 to 298
Data columns (total 13 columns):
 #   Column                    Non-Null Count  Dtype  
---  ------                    --------------  -----  
 0   age                       299 non-null    int32  
 1   anaemia                   299 non-null    bool   
 2   creatinine_phosphokinase  299 non-null    int64  
 3   diabetes                  299 non-null    bool   
 4   ejection_fraction         299 non-null    int64  
 5   high_blood_pressure       299 non-null    bool   
 6   platelets                 299 non-null    float64
 7   serum_creatinine          299 non-null    float64
 8   serum_sodium              299 non-null    int64  
 9   sex                       299 non-null    int64  
 10  smoking                   299 non-null    bool   
 11  time                      299 non-null    int64  
 12  patient_dead              299 non-null    bool   
dtypes: bool(5), float64(2), int32(1), int64(5)
memory usage: 19.1 KB


In [16]:
heart_failure_df.head(10)

age  anaemia  creatinine_phosphokinase  diabetes  ejection_fraction  high_blood_pressure  platelets  serum_creatinine  serum_sodium     sex  smoking  time  patient_dead
0  75   False    582                       False     20                 True                 265        1.9               130           Male    False    4     True        
1  55   False    7861                      False     38                 False                263        1.1               136           Male    False    6     True        
2  65   False    146                       False     20                 False                162        1.3               129           Male    True     7     True        
3  50   True     111                       False     20                 False                210        1.9               137           Male    False    7     True        
4  65   True     160                       True      20                 False                327        2.7               116           Female  False    8     True        
5  90   True     47                        False     40                 True                 204        2.1               132           Male    True     8     True        
6  75   True     246                       False     15                 False                127        1.2               137           Male    False    10    True        
7  60   True     315                       True      60                 False                454        1.1               131           Male    True     10    True        
8  65   False    157                       False     65                 False                263        1.5               138           Female  False    10    True        
9  80   True     123                       False     35                 True                 388        9.4               133           Male    True     10    True

Check for Nulls

In [18]:
heart_failure_df.isnull().sum()

age                         0
anaemia                     0
creatinine_phosphokinase    0
diabetes                    0
ejection_fraction           0
high_blood_pressure         0
platelets                   0
serum_creatinine            0
serum_sodium                0
sex                         0
smoking                     0
time                        0
patient_dead                0
dtype: int64